filters/filters.go

```diff
@@ -117,6 +117,19 @@ type FilterContext interface {
 
 	// Performs a new route lookup and executes the matched route if any
 	Loopback()
+
+	Logger() FilterContextLogger
+}
+
+// FilterContextLogger is the logger object which can log messages from filter
+// including context information from this filter.
+type FilterContextLogger interface {
+	Debugf(format string, args ...interface{})
+	Infof(format string, args ...interface{})
+	Warnf(format string, args ...interface{})
+	Errorf(format string, args ...interface{})
+	Println(args ...interface{})
+	Warningf(format string, args ...interface{})
```

Comment:
Lets remove these two. Warningf is a synonym for Warnf and we should use Debugf or Infof instead of Printf